### ISSUE TITLE
Expanding the model.

### ISSUE THREAD

**Issue by hristorv** [zhanghang1989/MSG-Net#5]
Greetings,
What are the capabilities, in terms of expanding the model ? What about 100 or even 10,000 styles in one model ? Will the quality of the pictures decline or stay the same if there are more styles in the model. Also how big will be a model containing that amount of pictures ? What are the possibilities of adding new styles to the existing model, lets say i have 100 styles model and i want to add 10 more, how will that happen?

**Comment by zhanghang1989** [comment]
Hi, that's a good question. Scalability is definitely an important thing. 

We have trained the MSG-NET for 100 styles without any problem, but definitely a much larger number (such as 10,000) will reach its limitation. I am working on an improved version of this work that can scale better, but haven't published yet. I would suggest exploring Adversarial loss instead of loss from pre-trained VGG. (a concurrent work 'AdaIN' can learns from thousands of styles and gets comparable results, but they haven't release the training) 

About incremental training, MSG-Net could not handle that. I would suggest the work of [style-bank](https://arxiv.org/pdf/1703.09210.pdf) (CVPR17).

**Comment by hristorv** [comment]
I have played with AdaIN and even worked on the Tensorflow implementation (still using the original model). As far as i can say your results seem similar to the older approaches in terms of quality. AdaIN lacks quality and i dont really know how those methods can be improved to achieve quality similar to 'Perceptual losses' or some of the other feed forward networks. 
I dont really know if its even possible to achieve such a quality. I think the biggest problem for style transfer right now is the trade off between, Quality - Speed - Flexibility.

**Comment by zhanghang1989** [comment]
@hristorv I agree. It is exactly the current difficulty of style transfer. Good luck to your work!
